ai_edge_torch/generative/layers/experimental/kv_cache.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2024 The AI Edge Torch Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+# Cache management utilities.
+
+from dataclasses import dataclass
+from typing import Any, Dict, List, Tuple
+
+import torch
+
+
+@dataclass
+class KVCache:
+  """A utility class for holding Key and Value caches per layer."""
+
+  k_caches: List[torch.Tensor]
+  v_caches: List[torch.Tensor]
+
+  def __init__(
+      self,
+      num_layers: int,
+      shape: Tuple[int],
```

Comment:
i guess eventually we might want a list of shapes or something, so each layer's kv shape can be different.